packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -1036,3 +1043,21 @@ const precisionRound = (value: number, precision: number) => {
   const factor = Math.pow(10, precision);
   return Math.round(value * factor) / factor;
 };
+
+const getLegendProps = (data: Data[], layout: Partial<Layout> | undefined) => {
+  const legends: string[] = [];
+  if (data.length === 1) {
+    legends.push(data[0].name || '');
```

Comment:
>'' [](http://example.com/codeflow?start=33&length=2)

Are we hiding the legend in case data[0].name is not present or empty string.

Reply:
To be taken later

Reply:
To keep the behavior similar to Plotly, legends are only hidden when the showlegend property is set to false. But, according to the [documentation](https://plotly.com/javascript/reference/layout/#layout-showlegend), legends are also hidden by default when there is only a single trace and showlegend is not explicitly set to true. I’ll follow up on this.